native/src/main/java/io/ballerina/stdlib/data/jsondata/FromString.java

```diff
@@ -0,0 +1,188 @@
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Native implementation of data:fromStringWithType(string).
+ *
+ * @since 0.1.0
+ */
+public class FromString {
+
+    private static final HashMap<Integer, Integer> TYPE_PRIORITY_ORDER = new HashMap<>() {{
+        int precedence = 0;
+        put(TypeTags.INT_TAG, precedence++);
+        put(TypeTags.FLOAT_TAG, precedence++);
+        put(TypeTags.DECIMAL_TAG, precedence++);
+        put(TypeTags.NULL_TAG, precedence++);
+        put(TypeTags.BOOLEAN_TAG, precedence++);
+        put(TypeTags.JSON_TAG, precedence++);
+        put(TypeTags.STRING_TAG, precedence);
+    }};
+
+    private static final ArrayList<Type> BASIC_JSON_MEMBER_TYPES = new ArrayList<>() {{
+        add(PredefinedTypes.TYPE_NULL);
+        add(PredefinedTypes.TYPE_BOOLEAN);
+        add(PredefinedTypes.TYPE_INT);
+        add(PredefinedTypes.TYPE_FLOAT);
+        add(PredefinedTypes.TYPE_DECIMAL);
+        add(PredefinedTypes.TYPE_STRING);
+    }};
+    private static final UnionType JSON_TYPE_WITH_BASIC_TYPES = TypeCreator.createUnionType(BASIC_JSON_MEMBER_TYPES);
+
+    public static Object fromStringWithType(BString string, BTypedesc typed) {
+        Type expType = typed.getDescribingType();
+
+        try {
+            return fromStringWithType(string, expType);
+        } catch (NumberFormatException e) {
+            return returnError(string.getValue(), expType.toString());
+        }
+    }
+
+    public static Object fromStringWithType(BString string, Type expType) {
+        String value = string.getValue();
+        try {
+            switch (expType.getTag()) {
+                case TypeTags.INT_TAG:
+                    return stringToInt(value);
+                case TypeTags.FLOAT_TAG:
+                    return stringToFloat(value);
+                case TypeTags.DECIMAL_TAG:
+                    return stringToDecimal(value);
+                case TypeTags.STRING_TAG:
+                    return string;
+                case TypeTags.BOOLEAN_TAG:
+                    return stringToBoolean(value);
+                case TypeTags.NULL_TAG:
+                    return stringToNull(value);
+                case TypeTags.UNION_TAG:
+                    return stringToUnion(string, (UnionType) expType);
+                case TypeTags.JSON_TAG:
+                    return stringToUnion(string, JSON_TYPE_WITH_BASIC_TYPES);
+                case TypeTags.TYPE_REFERENCED_TYPE_TAG:
+                    return fromStringWithType(string, ((ReferenceType) expType).getReferredType());
+                default:
+                    return returnError(value, expType.toString());
+            }
+        } catch (NumberFormatException e) {
+            return returnError(value, expType.toString());
+        }
```

Comment:
Can't we combine these lines and add something like this,

```suggestion
            }
            return returnError(value, expType.toString());
```

Reply:
Is the suggestion correct? we have to handle the exception right?